### PR TITLE
feat: make get_distance and recreate_collection public, replace deprecated recreate_collection function

### PR DESCRIPTION
**Pull request by ruben-vb**
### Proposed Changes:

As mentioned in the discord with @masci, making the get_distance and recreate_collection functions public. I also think the on_disk and use_sparse_embeddings could be optional parameters if there is no reason against this. 

I've also replaced the deprecated recreate_collection function and used the collection_exists check instead of the try-catch. Let me know if there is something that i've overlooked doing this.

### How did you test it?

unit tests, manual verification

### Checklist

- I have read the [contributors guidelines](https://github.com/deepset-ai/haystack-core-integrations/blob/main/CONTRIBUTING.md) and the [code of conduct](https://github.com/deepset-ai/haystack-core-integrations/blob/main/CODE_OF_CONDUCT.md)
- I've used one of the [conventional commit types](https://www.conventionalcommits.org/en/v1.0.0/) for my PR title: `fix:`, `feat:`, `build:`, `chore:`, `ci:`, `docs:`, `style:`, `refactor:`, `perf:`, `test:`.

### CODE REVIEW

**Review by masci — APPROVED** [review]
LGTM!

**Comment by masci** [comment]
@ruben-vb can you sign the CLA? I'll merge right after.

**Comment by ruben-vb** [comment]
> @ruben-vb can you sign the CLA? I'll merge right after.

done :)

**Pull request merged**
